controllers/componentdetectionquery_controller.go

```diff
@@ -145,23 +150,34 @@ func (r *ComponentDetectionQueryReconciler) Reconcile(ctx context.Context, req c
 						return ctrl.Result{}, nil
 					}
 
-					devfileBytes, err = devfile.DownloadDevfile(gitURL)
+					devfileBytes, dockerfileBytes, err = devfile.DownloadDevfileAndDockerfile(gitURL)
```

Comment:
we do this because this could be the four cases in a repo

1. Devfile & Dockerfile 
2. No Devfile & Dockerfile 
3. Devfile & No Dockerfile
4. No Devfile & No Dockerfile 

We read the dockerfile bascially for Case 2 when its a Dockerfile component.  I think its better to read both the files together when the repo is being scanned than scanning again after looking for a Devfile.